libs/R-TypeLogic/Global/Systems/NoAfkInMenuSystem.cpp

```diff
@@ -6,25 +6,24 @@
 */
 
 #include "NoAfkInMenuSystem.hpp"
-#include "R-TypeLogic/EntityManipulation/ButtonManipulation/SharedResources/MenuStates.hpp"
+#include <mutex>
+#include "R-TypeLogic/EntityManipulation/ButtonManipulation/SharedResources/GameStates.hpp"
 #include "R-TypeLogic/Global/Components/VelocityComponent.hpp"
 #include "R-TypeLogic/Global/SharedResources/GameClock.hpp"
 #include "R-TypeLogic/Server/Components/AfkFrequencyComponent.hpp"
-#include <mutex>
 
 using namespace ecs;
 
 void NoAfkInMenu::run(World &world)
 {
     std::vector<std::shared_ptr<Entity>> joined = world.joinEntities<AfkFrequency>();
 
-   MenuStates &menuState = world.getResource<MenuStates>();
-    menuState.lock();
-    MenuStates::menuState_e currState = menuState.currentState;
-    menuState.unlock();
-    if (currState == MenuStates::IN_GAME)
+    GameStates &gameState = world.getResource<GameStates>();
+    gameState.lock();
+    GameStates::gameState_e currState = gameState.currentState;
+    gameState.unlock();
```

Comment:
You can create a new scope like this
```
{
    std::lock_guard(gameState);
    GameStates::gameState_e currState = gameState.currentState;
}
```
instead of using lock and unlock.

Reply:
:eyes: